lib/ddtrace/span.rb

```diff
@@ -254,6 +254,61 @@ def to_hash
       h
     end
 
+    # MessagePack serializer interface. Making this object
+    # respond to `#to_msgpack` allows it to be automatically
+    # serialized by MessagePack.
+    #
+    # This is more efficient than doing +MessagePack.pack(span.to_hash)+
+    # as we don't have to create an intermediate Hash.
+    #
+    # @param packer [MessagePack::Packer] serialization buffer, can be +nil+ with JRuby
+    def to_msgpack(packer = nil)
```

Comment:
@Kyle-Verhoog can you review this?

Are there any optimizations you found from the Python implementation that we might be able to borrow here as well?